### PR TITLE
test: remove outdated todos in tests

### PR DESCRIPTION
**Pull request by ethanglaser**
## Description

Removes some TODOs in various parts of oneDAL testing as the issues are no longer leading to failed tests locally or in CI

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

**Comment by ethanglaser** [comment]
First job was green. will run another to check for any sporadic fails

**Comment by ethanglaser** [comment]
Adding PVC bazel test to private CI scope: http://intel-ci.intel.com/efb979ed-9041-f157-a732-a4bf010d0e2d

File: cpp/oneapi/dal/algo/kmeans/test/batch.cpp [hunk 1]
```diff
@@ -100,9 +100,6 @@ TEMPLATE_LIST_TEST_M(kmeans_batch_test,
                      "kmeans block test",
                      "[kmeans][batch][nightly][block]",
                      kmeans_types) {
-    // This test is not stable on CPU
-    // TODO: Remove the following `SKIP_IF` once stability problem is resolved
-    SKIP_IF(this->get_policy().is_cpu());
     SKIP_IF(this->is_sparse_method());
```
Comment (ethanglaser) [inline cpp/oneapi/dal/algo/kmeans/test/batch.cpp:103]:
```suggestion
    // This test is not stable on CPU
    // TODO: Remove the following `SKIP_IF` once stability problem is resolved
    SKIP_IF(this->get_policy().is_cpu());
    SKIP_IF(this->is_sparse_method());
```

**Comment by ethanglaser** [comment]
http://intel-ci.intel.com/efbc01b5-df63-f1e1-b8e8-a4bf010d0e2d

**Review by ahuber21 — APPROVED** [review]

**Pull request merged**